### DISCUSSION TITLE
Why does not a request pass through Middleware ?

### DISCUSSION THREAD

**Discussion by sabotenpillow** [honojs/hono#3487] [Q&A] [answered]
I want to check cookie in Middleware.
But it is not checked.

Sample code is here.

```typescript
import { Hono } from "hono"
import { serve } from "@hono/node-server";
import { createMiddleware } from "hono/factory";

const app = new Hono({ strict: false })

app.get("/service", (c) => {
  return c.json({ data: "foobar" });
});

const checkSession = createMiddleware(async (c, next) => {
  const sessionToken = getCookie(c, "session_token");
  if (sessionToken !== SessionToken) {
    c.status(401);
    return c.text("Unauthorized");
  }
  await next();
});

app.use("/service", checkSession);

const port = 3000;
console.log(`Server is running on port ${port}`);

serve({
  fetch: app.fetch,
  port,
});
```

And the log of `console.log` on Middleware is not outputted.

Why does not a request pass through Middleware ?

**Comment by safwanyp** [comment, marked as answer]
This is an issue with the ordering of your middlewares and handlers. 
Middlewares and handlers are executed in an "onion" layout. You can see that documented [here](https://hono.dev/docs/concepts/middleware#middleware) and [here](https://hono.dev/docs/guides/middleware#execution-order). Essentially, everything before the `await next()` in your middleware will be executed before the handler, and everything after the `await next()` will be executed after the handler.

In the sample code you provided, the middleware is registered _after_ the handler, which means that nothing it is technically never invoked. You can fix it like this:
```js
const checkSession = createMiddleware(async (c, next) => {
  const sessionToken = getCookie(c, "session_token");
  if (sessionToken !== SessionToken) {
    c.status(401);
    return c.text("Unauthorized");
  }
  await next();
});

// First, register the middleware
app.use("/service", checkSession);

// Second, register the handler
app.get("/service", (c) => {
  return c.json({ data: "foobar" });
});

const port = 3000;
console.log(`Server is running on port ${port}`);

serve({
  fetch: app.fetch,
  port,
});
```

**Comment by sabotenpillow** [comment]
Appreciate your response.
This problem solved.
Thank you!
